mne_sandbox/connectivity/tests/test_cfc.py

```diff
@@ -0,0 +1,180 @@
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Praveen Sripad <praveen.sripad@rwth-aachen.de>
+#          Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+import mne
+from itertools import combinations
+from nose.tools import assert_true, assert_raises, assert_equal
+from numpy.testing import assert_allclose
+from mne_sandbox.connectivity import (phase_amplitude_coupling,
+                                      phase_locked_amplitude,
+                                      phase_binned_amplitude)
+from sklearn.preprocessing import scale
+from mne.externals.pacpy import pac
+
+pac_func = 'plv'
+
+
+def _create_rand_data():
+    """Quickly create some random data."""
+    # Set params
+    sfreq = 1000.
+    n_sig = 3
+    n_ep = 8
+    n_t = 20.  # Time in one trial
+    ev = np.linspace(0, sfreq * n_t, n_ep).astype(int)
+    ev = np.vstack([ev, np.zeros_like(ev), np.ones_like(ev)]).T
+    t = np.linspace(0, n_t, sfreq * n_t)
+
+    ixs_conn = np.array(list(combinations(range(n_sig), 2)))
+    info = mne.create_info(n_sig, sfreq, 'eeg')
+
+    # Test random signals
+    ev = np.vstack([np.linspace(0, len(t), n_ep),
+                    np.zeros(n_ep),
+                    np.ones(n_ep)]).astype(int).T
+    rng = np.random.RandomState(42)
+    rand_data = rng.randn(n_ep, n_sig, len(t))
+    rand_raw = mne.io.RawArray(np.hstack(rand_data), info)
+    rand_epochs = mne.Epochs(rand_raw, ev, {'ev': 1}, -1, 8, preload=True)
+    return rand_raw, rand_epochs, ev, ixs_conn
+
+
+def test_phase_amplitude_coupling():
+    """ Test phase amplitude coupling. """
+    from scipy.signal import hilbert
+    flo = [4, 8]
+    fhi = [80, 150]
+    eptmin, eptmax = 1, 5
+
+    rand_raw, rand_epochs, ev, ixs_conn = _create_rand_data()
+    assert_raises(ValueError,
+                  phase_amplitude_coupling, rand_epochs, flo, fhi, ixs_conn)
+
+    rand_raw_test = rand_raw.crop(0, 15, copy=True)  # To speed things up
+    conn = phase_amplitude_coupling(
+        rand_raw_test, flo, fhi, ixs_conn, pac_func=pac_func)
+    assert_true(conn.mean() < .2)
+
+    # Test events handling
+    conn = phase_amplitude_coupling(rand_raw, flo, fhi, ixs_conn, ev=ev[:, 0],
+                                    tmin=eptmin, tmax=eptmax, pac_func=pac_func)
+    assert_true(conn.mean() < .2)
+    # events ndim > 1
+    assert_raises(ValueError, phase_amplitude_coupling, rand_raw, flo, fhi,
+                  ixs_conn, tmin=eptmin, tmax=eptmax, ev=ev, pac_func=pac_func)
+    # No tmin/tmax
+    assert_raises(ValueError, phase_amplitude_coupling, rand_raw, flo, fhi,
+                  ixs_conn, ev=ev, pac_func=pac_func)
+
+    # Test low frequency carrier / modulated oscillation
+    n_t = 10.
```

Comment:
if it's a an `n` it's generally a count, so int. if it's a duration, you'll should probably change the name